src/misc/lv_array.c

```diff
@@ -45,71 +45,143 @@ void lv_array_init(lv_array_t * array, uint32_t capacity, uint32_t element_size)
     LV_ASSERT_MALLOC(array->data);
 }
 
-void lv_array_destroy(lv_array_t * array)
+void lv_array_deinit(lv_array_t * array)
 {
     if(array->data) {
         lv_free(array->data);
         array->data = NULL;
     }
+
+    array->size = 0;
+    array->capacity = 0;
 }
 
-void lv_array_copy(lv_array_t * target, const lv_array_t * array)
+void lv_array_copy(lv_array_t * target, const lv_array_t * source)
 {
-    if(lv_array_is_empty(array)) {
+    if(lv_array_is_empty(source)) {
         return;
     }
-    lv_array_destroy(target);
-    lv_array_init(target, array->capacity, array->element_size);
-    lv_memcpy(target->data, array->data, array->size * array->element_size);
-    target->size = array->size;
+    lv_array_deinit(target);
+    lv_array_init(target, source->capacity, source->element_size);
+    lv_memcpy(target->data, source->data, source->size * source->element_size);
+    target->size = source->size;
 }
 
 void lv_array_clear(lv_array_t * array)
 {
     array->size = 0;
 }
 
-void lv_array_resize(lv_array_t * array, uint32_t new_capacity)
+/**
+ * Remove the element at the specified position in the array.
+ * @param array pointer to an `lv_array_t` variable
+ * @param index the index of the element to remove
+ */
+bool lv_array_remove(lv_array_t * array, uint32_t index)
+{
+    if(index >= array->size) {
+        return false;
+    }
+
+    /*Shortcut*/
+    if(index == array->size - 1) {
+        lv_array_resize(array, array->size - 1);
+        return false;
```

Comment:
In case the last item in the array is removed we can return `true`, because it succeeded.
But should we really resize the array after every remove? Or just after a given threshold (capacity - size >= THRESHOLD),

Reply:
It should return true. Fixed.

For performance, we can optimize them later with a THRESHOLD. I suspect it involves many APIs. Please open a new issue that we can follow up.